test/unit/test_shm_tensor.py

```diff
@@ -0,0 +1,39 @@
+import unittest
+import multiprocessing.shared_memory as shared_memory
+from tinygrad.helpers import CI
+from tinygrad.runtime.ops_shm import RawShmBuffer
+from tinygrad.tensor import Tensor, Device
+import numpy as np
+
+class TestRawShmBuffer(unittest.TestCase):
+  def test_e2e(self):
+    t = Tensor.randn(2, 2, 2).realize()
+
+    # copy to shm
+    shm_name = (s := shared_memory.SharedMemory(create=True, size=t.nbytes())).name
+    s.close()
+    t_shm = t.to(f"shm:{shm_name}").realize()
+
+    # copy from shm
+    t2 = t_shm.to(Device.DEFAULT).realize()
+
+    assert np.allclose(t.numpy(), t2.numpy())
+    s.unlink()
+
+  @unittest.skipIf(CI, "CI doesn't like big mmaps")
```

Comment:
Hmm, this isn't big, it's only 128MB. I really doubt it's the mmap, maybe it's the shm partition size?

Reply:
Yea, I'm not quite sure, but it does fail during the mmap, but it might be something like an mmap limit in actions?

Reply:
I think there's `SHMMAX` and `SHMALL` curious what they are on CI

Reply:
Don't think that's it, cuz it reports the same on github actions as on my machine:

```
 SHMMNI   Shared memory segments                                    4096    0 0.00%
SHMALL   Shared memory pages                       18446744073692774399    0 0.00%
SHMMAX   Max size of shared memory segment (bytes)                  16E    -     -
SHMMIN   Min size of shared memory segment (bytes)                   1B    -     -
```